jerry-core/ecma/base/ecma-helpers-string.c

```diff
@@ -135,18 +134,19 @@ ecma_string_get_chars_fast (const ecma_string_t *string_p, /**< ecma-string */
   {
     if (ECMA_GET_DIRECT_STRING_TYPE (string_p) == ECMA_DIRECT_STRING_MAGIC)
     {
-      lit_magic_string_id_t id = (lit_magic_string_id_t) ECMA_GET_DIRECT_STRING_VALUE (string_p);
+      uint32_t id = (uint32_t) ECMA_GET_DIRECT_STRING_VALUE (string_p);
```

Comment:
We need to document somewhere that we don't support UINT32_MAX number of magic strings anymore. It is impractical anyway, but we need to document it. We could say at least `UINT32_MAX / 2` is supported, maybe more. Also, if set external magic strings is called, and the number of magic strings is greater than the limit, we should "crop" the rest:
```
if (number_of_ext_magic_string > UINT32_MAX - MAGIC_STRINGS)
   number_of_ext_magic_string = UINT32_MAX - MAGIC_STRINGS
```

Reply:
Good idea!

Reply:
I'd vote for the `UINT32_MAX/2` since `MAGIC_STRING_COUNT` is not a constant value.